hippod/api_object_get_full.py

```diff
@@ -72,50 +71,70 @@ def get_last_attachment_data(app, sha_sum, cont_obj):
     return r
 
 
-def container_obj_to_ret_obj(app, sha_sum, cont_obj):
+def container_obj_to_ret_obj(app, sha_major, cont_obj):
     ret_obj = dict()
-
     # add object item ID
-    ret_obj['object-item-id'] = sha_sum
+    ret_obj['object-item-id'] = sha_major
 
     # add some object items
     ret_obj['object-item'] = dict()
-    ret_obj['object-item']['title'] = cont_obj['object-item']['title'] 
-    ret_obj['object-item']['version'] = cont_obj['object-item']['version']
-    if 'data' in cont_obj['object-item']:
-        ret_obj['object-item']['data'] = cont_obj['object-item']['data']
-
-    ret_obj['maturity-level'] = cont_obj['maturity-level'][-1]
+    ret_obj['object-item']['title'] = cont_obj['title']
+    ret_obj['object-item']['version'] = cont_obj['version']
+    ret_obj['subcontainer'] = list()
+
+    for sub_cont in cont_obj['subcontainer-list']:
+        sub_dict = dict()
+        sub_dict['object-item'] = dict()
+        sub_dict['sha_minor'] = sub_cont['sha_minor']
+        sub_dict['object-item']['data'] = list()
+        ok, full_sub_cont = hippod.api_shared.read_subcont_obj_by_id(app, sha_major, sub_cont['sha_minor'])
+        if ok:
+            data = get_all_achievement_data(app, sha_major, sub_cont['sha_minor'], full_sub_cont)
+            if data:
+                sub_dict['object-achievements'] = data
+            if 'data' in full_sub_cont['object-item']:
+                print(sub_dict)
+                sub_dict['object-item']['data'].append(full_sub_cont['object-item']['data'])
+            # exception not required?
+        else:
```

Comment:
the error handling is good but not perfect, prefer the following:

```
if not ok:
    raise Error()
thing_goes_on()
no_error_case()
```

compared to 

```
if ok:
    thing_goes_on()
    no_error_case()
else:
    raise Error()
```